shared/gh/js/views/gh.admin-batch-edit.js

```diff
@@ -1205,12 +1214,71 @@ define(['gh.core', 'gh.constants', 'gh.utils', 'moment', 'gh.calendar', 'gh.admi
                     // TODO: Remove this and only trigger when button is clicked/expanded
                     $(document).trigger('gh.batchdate.setup');
                     $(document).trigger('gh.batchorganiser.setup');
-                });
+                };
+
+                // Fetch the part the borrowed series belongs to
+                if (series.borrowedFrom) {
+                    gh.api.orgunitAPI.getOrgUnit(series.borrowedFrom.ParentId, false, function(err, part) {
+                        if (err) {
+                            return utils.notification('Part data not retrieved.', 'The part data could not be successfully retrieved.', 'error');
+                        }
+
+                        // Fetch the tripos the borrowed series belogs to
+                        if (part.ParentId) {
+                            gh.api.orgunitAPI.getOrgUnit(part.ParentId, false, function(err, tripos) {
+                                if (err) {
+                                    return utils.notification('Tripos data not retrieved.', 'The tripos data could not be successfully retrieved.', 'error');
+                                }
+
+                                // Get the first page of events
+                                getSeriesEvents(0, onGetSeriesEventsComplete, {
+                                    'part': part,
+                                    'tripos': tripos
+                                });
+                            });
+                        }
+                    });
+
+                // Immediately get the first page of events when the series is not borrowed
+                } else {
+                    getSeriesEvents(0, onGetSeriesEventsComplete);
+                }
             });
         }
     };
 
 
+    ///////////////
+    // BORROWING //
+    ///////////////
+
+    /**
+     * Update the tripos in the url
+     *
+     * @private
+     */
+    var updateTripos = function() {
+        var $trigger = $(this);
+        var triposId = $trigger.data('id');
+        utils.removeFromState(['tripos', 'part', 'module', 'series']);
+        utils.addToState({'tripos': triposId});
+    };
+
+    /**
+     * Update the part in the url
+     *
+     * @private
+     */
+    var updatePart = function() {
+        var $trigger = $(this);
+        var triposId = $trigger.data('parentid');
+        var partId = $trigger.data('id');
+        utils.removeFromState(['tripos', 'part', 'module', 'series']);
```

Comment:
And I think this'll need to be run after the `addToState` call because the removeState util replaces the history entry behind the scenes and running this before addToState will cause issues.